packages/editor/src/components/post-title/index.native.js

```diff
@@ -91,27 +87,43 @@ class PostTitle extends Component {
 	}
 }
 
-const applyWithDispatch = withDispatch( ( dispatch ) => {
-	const {
-		undo,
-		redo,
-	} = dispatch( 'core/editor' );
+export default compose(
+	withSelect( ( dispatch ) => {
+		const {
+			isPostTitleSelected,
+		} = dispatch( 'core/editor' );
 
-	const {
-		insertDefaultBlock,
-	} = dispatch( 'core/block-editor' );
+		return {
+			isSelected: isPostTitleSelected(),
+		};
+	} ),
+	withDispatch( ( dispatch ) => {
+		const {
+			undo,
+			redo,
+			togglePostTitleSelection,
+		} = dispatch( 'core/editor' );
 
-	return {
-		onEnterPress() {
-			insertDefaultBlock( undefined, undefined, 0 );
-		},
-		onUndo: undo,
-		onRedo: redo,
-	};
-} );
+		const {
+			clearSelectedBlock,
+			insertDefaultBlock,
+		} = dispatch( 'core/block-editor' );
 
-export default compose(
-	applyWithDispatch,
+		return {
+			onEnterPress() {
+				insertDefaultBlock( undefined, undefined, 0 );
+			},
+			onUndo: undo,
+			onRedo: redo,
+			onSelect() {
+				togglePostTitleSelection( true );
+				clearSelectedBlock();
```

Comment:
Asking more for my understanding: is it worth considering handling the `clearSelectedBlock` call inside a reducer for the `TOGGLE_POST_TITLE_SELECTION` action anytime it is set to `true`? My thinking is just that we would _always_ want to clear any other selected block anytime we set the post title to be selected, and handling it in a reducer would avoid us needing to remember to always dispatch these two actions together. I do not feel strongly about this, but I would be interested in hearing any reasons for or against this in your mind.

Reply:
A reducer should only ever update the state so I don't think it would be appropriate. An action could be calling another action but I think it's more explicit to have it here, especially since it's a different store!

Reply:
Good point about the reducer just updating the store, and I totally overlooked the different store. Thanks! 👍